### PR TITLE
actually unmarshal decorate configs

### PR DESCRIPTION
**Pull request by krzyzacy**
:upside_down_face: only took me an hour to figure out

/area prow
/assign @cjwagner @stevekuznetsov @BenTheElder

### CODE REVIEW

**Comment by krzyzacy** [comment]
ref https://github.com/kubernetes/test-infra/pull/9564#issuecomment-424808031, obviously I'm the first one trying to do this

**Comment by krzyzacy** [comment]
hummmm I saw the integration test now

so looks like I should just be applying fields directly

/hold

**Comment by krzyzacy** [comment]
(I think this is more correct, hold for discussion)

**Comment by stevekuznetsov** [comment]
Oops. Shouldn't that be 

    *kube.DecorationConfig `json:",inline"`

**Comment by krzyzacy** [comment]
I feel like this should match https://github.com/kubernetes/test-infra/blob/master/prow/apis/prowjobs/v1/types.go#L140
```
	// DecorationConfig holds configuration options for
	// decorating PodSpecs that users provide
	DecorationConfig *DecorationConfig `json:"decoration_config,omitempty"`
```

**Comment by cjwagner** [comment]
I agree that it is a bit confusing to a developer that the ProwJob structures the data differently than the config object, but I think keeping the fields inlined in the config object makes configuration a little simpler.
The normal user of Prow will never look at how the decoration configuration is stored in a PJ, but they will configure jobs.

I don't feel too strongly about this though.

**Comment by BenTheElder** [comment]
I don't think we should serialize the prowjob differently from the config other than derived fields, fwiw.

**Comment by stevekuznetsov** [comment]
I have no strong feelings either way -- sorry about the oversight

**Comment by cjwagner** [comment]
We could also change ProwJob to embed the struct pointer like the UtilityConfig does.
/shrug

**Comment by BenTheElder** [comment]
sgtm, I don't think it's a big deal but we should probably pick one and align them to match.

**Comment by krzyzacy** [comment]
(I prefer not to touch prowjobs... seems at least nobody is setting explicit decorations in our config, I'll go with this and fix integration tests)

**Comment by krzyzacy** [comment]
/hold cancel
(also removed more generated fields from test data)

File: prow/config/jobs.go [hunk 1]
```diff
@@ -329,7 +329,7 @@ type UtilityConfig struct {
 
 	// DecorationConfig holds configuration options for
 	// decorating PodSpecs that users provide
-	*kube.DecorationConfig
+	*kube.DecorationConfig `json:"decoration_config,omitempty"`
```
[the remark this answers is not in the document]
Comment (krzyzacy) [reply prow/config/jobs.go]: Change prowjobs will cause issues with older versions of mkpj, anything else might break?

**Comment by krzyzacy** [comment]
> this is why we need yaml string unit tests.

I'll add one :upside_down_face:

File: prow/config/jobs.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (fejta) [reply prow/config/jobs.go]: Do we need/want to have an embedded type if we're not going to embed the fields in the yaml?

[the remark this answers is not in the document]
Comment (krzyzacy) [reply prow/config/jobs.go]: sorry I didn't follow, what's the alternative?

[the remark this answers is not in the document]
Comment (fejta) [reply prow/config/jobs.go]: This is still am embedded field. 

Compare the type signature in the prowjob:

```
	// DecorationConfig holds configuration options for
	// decorating PodSpecs that users provide
	DecorationConfig *DecorationConfig `json:"decoration_config,omitempty"`
```

They are not equivalent:

`DecorationConfig *DecorationConfig` (prowjob) != `*kube.DecorationConfig` (presubmit, postsubmit, periodic)


So the alternative here is to give `*kube.DecorationConfig` a name (like DecorationConfig) so it isn't embedded anymore.

[the remark this answers is not in the document]
Comment (krzyzacy) [reply prow/config/jobs.go]: thanks for explanation ^^ will do that

[the remark this answers is not in the document]
Comment (krzyzacy) [reply prow/config/jobs.go]: done

File: prow/ANNOUNCEMENTS.md [hunk 2]
```diff
@@ -25,9 +25,43 @@ Note: versions specified in these announcements may not include bug fixes made
 in more recent versions so it is recommended that the most recent versions are
 used when updating deployments.
 
+ - *Sep 27, 2018* If you are setting explicit decorate configs, the format has changed from
```
Comment (fejta) [inline prow/ANNOUNCEMENTS.md]: I think this message is entirely too long? The only relevant part is needing to embed fields like timeout inside a `decoration_config` field.

Comment (krzyzacy) [reply prow/ANNOUNCEMENTS.md]: trimmed other fields.

**Comment by krzyzacy** [comment]
PTAL :-)

**Comment by krzyzacy** [comment]
oh it's Friday, bump time :-)
/hold cancel

**Pull request merged**